src/transition.jl

```diff
@@ -47,13 +47,21 @@ end
 POMDPs.support(d::LTTransDist) = d
 
 function Base.iterate(d::LTTransDist, i::Int=1)
-    if i > 5 || d.terminal && i > 1
+    if d.terminal
+        if i > 1
+            return nothing
+        else
+            return (LTState(d.rob, d.prev_opp, true), i+1)
+        end
+    end
+    while i <= 5 && d.probs[i] == 0.0
```

Comment:
With this while loop, I think `iterate` and `length` become inconsistent. This can lead to some strange behavior, for example:
```
julia> struct T end

julia> Base.length(::T) = 5

julia> function Base.iterate(::T, i=1)
           if i > 2
               return nothing
           else
               return (i, i+1)
           end
       end

julia> collect(T())
5-element Vector{Any}:
   1
   2
 #undef
 #undef
 #undef
```
I think we should either change `length` to remove the zero probability items, or just get rid of the while loop (it probably doesn't save that much).

Reply:
Zero probability items contain illegal states when `prev_opp` is at the border. When an illegal state is passed to `stateindex`, an error will occur. For example,
```julia
ERROR: LoadError: BoundsError: attempt to access 11×7×11×7 LinearIndices{4, NTuple{4, Base.OneTo{Int64}}} at index [1, 2, 1, 0]
Stacktrace:
  [1] throw_boundserror(A::LinearIndices{4, NTuple{4, Base.OneTo{Int64}}}, I::NTuple{4, Int64})
    @ Base .\abstractarray.jl:651
  [2] checkbounds
    @ .\abstractarray.jl:616 [inlined]
  [3] _getindex
    @ .\abstractarray.jl:1196 [inlined]
  [4] getindex
    @ .\abstractarray.jl:1170 [inlined]
  [5] stateindex(f::LaserTag.Floor, s::LTState)
    @ LaserTag ~\OneDrive\Workspace\julia\LaserTag.jl\src\states.jl:28
```
So, I think we should change `length`.